src/main/java/uk/nhs/digital/nhsconnect/lab/results/inbound/InboundMessageHandler.java

```diff
@@ -27,38 +29,62 @@
     private final EdifactToFhirService edifactToFhirService;
     private final EdifactParser edifactParser;
     private final GpOutboundQueueService gpOutboundQueueService;
-    private final RecepProducerService recepProducerService;
     private final MeshOutboundQueueService meshOutboundQueueService;
+    private final OutboundMeshMessageBuilder outboundMeshMessageBuilder;
 
     public void handle(final InboundMeshMessage meshMessage) {
-
-        final Interchange interchange = edifactParser.parse(meshMessage.getContent());
+        final Interchange interchange;
+        try {
+            interchange = edifactParser.parse(meshMessage.getContent());
+        } catch (InterchangeParsingException ex) {
+            LOGGER.error("Error parsing Interchange", ex);
+            var nhsack = outboundMeshMessageBuilder.buildNhsAck(meshMessage.getWorkflowId(), ex);
+            meshOutboundQueueService.publish(nhsack);
+            return;
+        } catch (MessagesParsingException ex) {
+            LOGGER.error("Error parsing Messages", ex);
+            var nhsack = outboundMeshMessageBuilder.buildNhsAck(meshMessage.getWorkflowId(), ex);
+            meshOutboundQueueService.publish(nhsack);
+            return;
+        }
 
         logInterchangeReceived(interchange);
 
-        final List<Message> messages = interchange.getMessages();
+        final List<MessageProcessingResult> messageProcessingResults = getFhirDataToSend(interchange.getMessages());
 
-        LOGGER.info("Interchange contains {} new messages", messages.size());
+        var nhsack = outboundMeshMessageBuilder.buildNhsAck(
+            meshMessage.getWorkflowId(),
+            interchange,
+            messageProcessingResults);
 
-        final List<Bundle> bundles = getBundlesToSend(messages);
+        messageProcessingResults.stream()
+            .filter(result -> result instanceof SuccessMessageProcessingResult)
+            .map(SuccessMessageProcessingResult.class::cast)
+            .map(SuccessMessageProcessingResult::getBundle)
+            .forEach(gpOutboundQueueService::publish);
 
-        bundles.forEach(gpOutboundQueueService::publish);
+        logSentFor(interchange);
 
-        //TODO temporarily disabled
-        //sendRecep(interchange);
+        meshOutboundQueueService.publish(nhsack);
 
-        logSentFor(interchange);
+        logRecepSentFor(interchange);
     }
 
-    private List<Bundle> getBundlesToSend(List<Message> messagesToProcess) {
-        return messagesToProcess.stream()
-            .map(message -> {
-                final Bundle bundle = edifactToFhirService.convertToFhir(message);
-
-                LOGGER.debug("Converted edifact message into {}", bundle);
+    private List<MessageProcessingResult> getFhirDataToSend(List<Message> messages) {
+        return messages.stream().parallel()
+            .map(this::convertToFhir)
+            .collect(Collectors.toList());
```

Comment:
Might be worth microbenchmarking this to verify the value; I've read the benefits of parallel streams only really kick in when processing 1000s+ of elements.